### ISSUE TITLE
Error in classifying multi class documents

### ISSUE THREAD

**Issue by vinhqdang** [tflearn/tflearn#515]
Hello everyone,

I am trying to follow the example at https://github.com/tflearn/tflearn/blob/master/examples/nlp/dynamic_lstm.py

I have documents stored in the directory ``text``, each file corresponds to one document, named ``1``, ``2`` ... There are 10 000 files.

I have a label file named ``labels``, contains 10 000 lines, each line is a label of each file. 

There are 6 labels, ``a``, ``b`` ... ``f``.

Here is my code:

```{python}
from __future__ import division, print_function, absolute_import

import tflearn
from tflearn.data_utils import to_categorical, pad_sequences

import csv
import numpy as np
from sklearn import metrics, cross_validation

import tensorflow as tf

import random
import time

import string

import getopt
import sys
import os

data_dir = "text"   
model_size = 2000  
nb_epochs      = 50   
label_file = "labels"
MAX_FILE_ID = 50000

### Training data

labels = ["a","b","c","d","e","f"]

print('Read labels')

def load_label (label_file):
    with open (label_file) as f:
        return f.read().splitlines()

Y = load_label(label_file)

for i in range(len(Y)):
    Y[i] = labels.index(Y[i])

print('Read content')

def load_content (file_name):
    with open(file_name) as f:
        return f.read()

X = []
for i in range (MAX_FILE_ID):
    file_name = data_dir + '/' + str(i + 1)
    if os.path.isfile (file_name):
        X.append (load_content(file_name)) 

X_train, X_test, Y_train, Y_test = cross_validation.train_test_split(X, Y,
    test_size=0.2, random_state=2017)

Y_train = to_categorical (Y_train, nb_classes = len(labels))
Y_test = to_categorical (Y_test, nb_classes = len(labels))

### Process vocabulary

print('Process vocabulary')

vocab_processor = tflearn.data_utils.VocabularyProcessor(max_document_length = model_size, min_frequency = 0)
X_train = np.array(list(vocab_processor.fit_transform(X_train)))
X_test = np.array(list(vocab_processor.fit_transform(X_test)))

X_train = pad_sequences(X_train, maxlen=model_size, value=0.)
X_test = pad_sequences(X_test, maxlen=model_size, value=0.)

n_words = len(vocab_processor.vocabulary_)
print('Total words: %d' % n_words)

### Models

print('Build model')

net = tflearn.input_data([None, model_size])
net = tflearn.embedding(net, input_dim=model_size, output_dim=128)
net = tflearn.lstm(net, 128, dropout=0.5)
net = tflearn.fully_connected(net, n_units = len(labels) , activation='softmax')
net = tflearn.regression(net, optimizer='adam', learning_rate=0.001,
                         loss='categorical_crossentropy')

print ('Train model')

model = tflearn.DNN(net, tensorboard_verbose=3)

print ('Predict')
model.fit(X_train, Y_train, validation_set=(X_test, Y_test), show_metric=True,
          batch_size=32)
```

and I met an error

``InvalidArgumentError (see above for traceback): indices[0,0] = 4683 is not in [0, 2000)``

**Comment by aymericdamien** [comment]
It seems there is an error with your data shape. Can you send full stack of the error to identify the line that causes it?

**Comment by vinhqdang** [comment]
I think the problem is the line:

``net = tflearn.embedding(net, input_dim=model_size, output_dim=128)``

it should be

``net = tflearn.embedding(net, input_dim=n_words, output_dim=128)``
